frontend/src/components/common/AuthModal/AuthModal.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback, useState } from 'react';
+import {
+  Box,
+  Button,
+  Link,
+  Modal,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { useLogin, useSignUp } from '@/hooks';
+import { loginConfig, signUpConfig, style } from './utils';
+import AuthErrorAlert from './AuthErrorAlert';
+import LoadingModal from '../LoadingModal';
+
+interface AuthModalProps {
+  onClose: () => void;
+  onSubmit: () => void;
+  isSignUpModal?: boolean;
+}
+
+export const AuthModal: React.FC<AuthModalProps> = ({
+  onClose,
+  onSubmit,
+  isSignUpModal = false,
+}) => {
+  const [isSignUp, setIsSignUp] = useState<boolean>(isSignUpModal);
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const modalConfig = isSignUp ? signUpConfig : loginConfig;
+
+  const [signUp, { isLoading: isSignUpLoading, error: signUpError }] =
+    useSignUp();
+  const [login, { isLoading: isLoginLoading, error: loginError }] = useLogin();
+
+  const isLoading = isSignUpLoading || isLoginLoading;
+  const error = signUpError || loginError;
+  const performAuth = isSignUp ? signUp : login;
+
+  const onPrimaryButtonClick = useCallback<() => Promise<void>>(async () => {
+    await performAuth({
+      email,
+      password,
+    });
+    !error && onSubmit();
+  }, [error, onSubmit, email, password, performAuth]);
+
+  if (isLoading) return <LoadingModal />;
+
+  return (
+    <Modal open onClose={onClose}>
```

Comment:
probably gonna be a pain to change now since you've done all the work, but if not should we suss the MUI dialog component instead for this? A lot of the styling is handled and imo looks a bit cleaner @KevinLu9 @Eeyore6 I've used it in my dashboard PR for a basic yes/no alert modal https://mui.com/material-ui/react-dialog/

Reply:
Definitely up for the Dialog change, but this card was more about the supabase functionality. (ie signup/login).
We'll make a card later that fleshes out the UI flow (after a discussion).